File: cli/python/rewriter.py
"""
Markdown+ rewriter — thin wrapper around OpenAI Responses API.

Usage:
    set OPENAI_API_KEY=sk-...
    python rewriter.py --from markdown input.md > output.mdp.md
    python rewriter.py --from html     input.html > output.mdp.md
    cat input.md | python rewriter.py --from markdown - > output.mdp.md

Options:
    --model         default: gpt-5.5
    --effort        reasoning.effort, default: none
"""
from __future__ import annotations

import argparse
import os
import sys
from pathlib import Path

from openai import OpenAI

ROOT = Path(__file__).resolve().parent
PROMPTS = {
    "markdown": ROOT / "prompts" / "from_markdown.txt",
    "html": ROOT / "prompts" / "from_html.txt",
}


def build_prompt(kind: str, content: str) -> str:
    tmpl = PROMPTS[kind].read_text(encoding="utf-8")
    placeholder = "{{markdown_doc}}" if kind == "markdown" else "{{html_doc}}"
    return tmpl.replace(placeholder, content)


def main() -> int:
    p = argparse.ArgumentParser()
    p.add_argument("input", help="input file path or '-' for stdin")
    p.add_argument("--from", dest="kind", choices=["markdown", "html"], required=True)
    p.add_argument("--model", default=os.environ.get("REWRITER_MODEL", "gpt-5.5"))
    p.add_argument("--effort", default=os.environ.get("REWRITER_EFFORT", "none"))
    args = p.parse_args()

    if not os.environ.get("OPENAI_API_KEY"):
        print("ERROR: OPENAI_API_KEY not set", file=sys.stderr)
        return 1

    if args.input == "-":
        content = sys.stdin.read()
    else:
        content = Path(args.input).read_text(encoding="utf-8")

    prompt = build_prompt(args.kind, content)

    client = OpenAI()
    response = client.responses.create(
        model=args.model,
        input=prompt,
        reasoning={"effort": args.effort},
    )
    sys.stdout.write(response.output_text or "")
    return 0


if __name__ == "__main__":
    sys.exit(main())

File: cli/node/rewriter.mjs
// Markdown+ rewriter — Node.js wrapper around OpenAI Responses API.
// Usage:
//   OPENAI_API_KEY=sk-... node rewriter.mjs --from markdown input.md > output.mdp.md
//   OPENAI_API_KEY=sk-... node rewriter.mjs --from html     input.html > output.mdp.md
//   cat input.md | OPENAI_API_KEY=sk-... node rewriter.mjs --from markdown - > output.mdp.md
//
// Requires:  npm install openai

import fs from "node:fs";
import path from "node:path";
import { fileURLToPath } from "node:url";
import OpenAI from "openai";

const __dirname = path.dirname(fileURLToPath(import.meta.url));

function parseArgs(argv) {
  const args = { input: null, from: null, model: process.env.REWRITER_MODEL || "gpt-5.5", effort: process.env.REWRITER_EFFORT || "none" };
  for (let i = 2; i < argv.length; i++) {
    const a = argv[i];
    if (a === "--from") args.from = argv[++i];
    else if (a === "--model") args.model = argv[++i];
    else if (a === "--effort") args.effort = argv[++i];
    else if (a.startsWith("--")) throw new Error(`Unknown flag ${a}`);
    else args.input = a;
  }
  if (!args.from || !args.input) {
    console.error("usage: node rewriter.mjs --from markdown|html <input|->");
    process.exit(2);
  }
  return args;
}

function readInput(p) {
  if (p === "-") {
    return fs.readFileSync(0, "utf-8");
  }
  return fs.readFileSync(p, "utf-8");
}

function buildPrompt(kind, content) {
  const tmplPath = path.join(__dirname, "prompts", kind === "markdown" ? "from_markdown.txt" : "from_html.txt");
  const tmpl = fs.readFileSync(tmplPath, "utf-8");
  const placeholder = kind === "markdown" ? "{{markdown_doc}}" : "{{html_doc}}";
  return tmpl.split(placeholder).join(content);
}

async function main() {
  if (!process.env.OPENAI_API_KEY) {
    console.error("ERROR: OPENAI_API_KEY not set");
    process.exit(1);
  }
  const args = parseArgs(process.argv);
  const content = readInput(args.input);
  const prompt = buildPrompt(args.from, content);

  const client = new OpenAI();
  const response = await client.responses.create({
    model: args.model,
    input: prompt,
    reasoning: { effort: args.effort },
  });
  process.stdout.write(response.output_text || "");
}

main().catch(e => { console.error(e); process.exit(1); });
